packages/web/docs/src/pages/self-hosting/get-started.mdx

```diff
@@ -89,23 +95,151 @@ the environment variables.
+step, but if not so we're gonna initialize a new package in our directory, you can use your
+preferred package manager to do so:
+
+```bash
+# you can do the same with npm or yarn
+pnpm init
+```
+
+Then create a `schema.graphql` file wherever you want, you can put it at the root for example with
+any simple graphql schema like the following:
+
+```graphql
+type Query {
+  hello: String!
+}
+```
+
+You also need to have a git repository initilaized for your project, because Hive automatically uses
+the last commit hash to determine who the author is and ties the schema with the commit hash. So
+we're gonna do that.
+
+```bash
+git init
+```
+
+Make sure you exclude the following in your `.gitignore` file, because you don't wanna push those in
+your commits:
+
+```
+node_modules
+.hive
+hive.json
+.env
+```
+
+Make sure to at least have a single commit so that Hive can associate the schema with it.
+
+```bash
+git add .
+git commit -m"init"
+```
+
+Now that you've your Nodejs project setup, start by installing the GraphQL Hive CLI which we're
+gonna use to interact with Hive's server.
+
+```bash
+# you can do the same with npm or yarn
+pnpm i -D @graphql-hive/cli
+```
+
+<Callout type="info">
+  We recommend that you install the GraphQL Hive CLI as a dev dependency
+</Callout>
+
+After installing it, the CLI's bin is set to `hive` which means you can check if it was installed by
+using this command:
+
+```bash
+# you can do the same with npm or yarn
+pnpm hive --version
+```
+
+Make sure you get an output specifiying the installed version for your CLI.
+
+And since we're using the Self Hosted version, we need to configure the Hive CLI to use our local
+endpoint, by creating a `hive.json` file at the root of your directory. Then specify the `registry`
+key to point it our local registry instead of the cloud version one and the value is gonna be
+`http://localhost:8082/graphql` where its running.
+
+Additionally the schemas registry only allows trusted sources to be able to push to them, so we have
+to create an access token by heading back to our project page, specifically on the target(in this
+guide it's `development`) then go to the settings tab, if you scroll donw a bit you're gonna see the
+`Tokens` section, click the `Generate New Token` button, give it a name then we'll have to set some
+permissions for that token, so expand the `All targets` section then give the `Registry` a
+`Read & Write` access. Click the `Generate Token` button, then copy the token value. Head back to
+the `hive.json` config file then add a new key `token` and paste the value from your clipboard.
+
+Now we're all setup to use the GraphQL Hive CLI against our Setup!
+
+We can publish our first schema by using:
+
+```bash
+pnpm hive schema:publish ./schema.graphql
+```
+
+Go to the `Schema` tab in your project target and you're gonna see your latest published schema.
+
 ## Next Steps
```

Comment:
Maybe we should have a section here to describe all the exposed ports and what's their purpose?

Reply:
oh yeah, definitely. I was confused myself by the ports available and what their purposes are